### PR TITLE
Anvil: Checksum address output in start-up log

### PR DESCRIPTION
**Pull request by purplesmoke05**
## Motivation

#4157 

## Solution

- Changed account addresses in start up log to checksummed with ethers::core::utils::to_checksum.

### CODE REVIEW

File: anvil/src/config.rs [hunk 1]
```diff
@@ -172,7 +172,8 @@ Available Accounts
         );
         let balance = format_ether(self.genesis_balance);
         for (idx, wallet) in self.genesis_accounts.iter().enumerate() {
-            let _ = write!(config_string, "\n({idx}) {:?} ({balance} ETH)", wallet.address());
+            let _ = write!(config_string, "\n({idx}) {:?} ({balance} ETH)", to_checksum(&wallet.address(), self.chain_id.and_then(|id| Some(u8::try_from(id).unwrap_or_default()))
```
Comment (mattsse) [inline anvil/src/config.rs:175]: this should be called with None, I believe, see `to_checksum` impl

I don't think we want the chain id prefix?

Comment (purplesmoke05) [reply anvil/src/config.rs]: Thanks for the review. XD I now understand what you pointed out.
Fixed in https://github.com/foundry-rs/foundry/pull/4158/commits/4bd34f8dead97d2122a0b168f970bdcb4a263230

**Review by mattsse — APPROVED** [review]
ty, lgtm

**Comment by jchaz22** [comment]
What's the work around, I'm getting a check sum error when pasting a deployed contract from my anvil into a different script?

**Pull request merged**